source/config/500-controller.yaml

```diff
@@ -43,7 +43,7 @@ spec:
         - name: CONFIG_OBSERVABILITY_NAME
           value: config-observability
         - name: METRICS_DOMAIN
-          value: knative.dev/sources
+          value: '{"Domain":"knative.dev/eventing","Component":"redisstreamsource","ConfigMap":{"profiling.enable": "true"}}'
```

Comment:
The controller gets the metric configuration directly from config-observability. On the other hand, the receive adapter expects the metric configuration to be inlined inside a env var (K_METRICS_CONFIG). Bottom line: keep METRICS_DOMAIN the same, and make sure the controller properly sets  K_METRICS_CONFIG

Reply:
Ok, I reverted METRICS_DOMAIN and will plan on fixing the controller's setup for K_METRICS_CONFIG in a separate commit outside of this PR (since I don't have it working yet).